test/scenarios/datafactory/output/src/datafactory-preview/azext_datafactory_preview/generated/custom.py

```diff
@@ -53,14 +53,22 @@ def datafactory_create(client,
                        'repo_configuration!')
     repo_configuration = all_repo_configuration[0] if len(all_repo_configuration) == 1 else None
     factory = {}
-    factory['location'] = location
-    factory['tags'] = tags
-    factory['test_inherit'] = test_inherit
-    factory['repo_configuration'] = repo_configuration
-    factory['fake_identity'] = fake_identity
-    factory['zones'] = zones
+    if location is not None:
+        factory['location'] = location
+    if tags is not None:
+        factory['tags'] = tags
+    if test_inherit is not None:
+        factory['test_inherit'] = test_inherit
+    if repo_configuration is not None:
+        factory['repo_configuration'] = repo_configuration
+    if fake_identity is not None:
+        factory['fake_identity'] = fake_identity
+    if zones is not None:
+        factory['zones'] = zones
     factory['identity'] = {}
     factory['identity']['type'] = "SystemAssigned"
+    if len(factory['identity']) == 0:
+        del factory['identity']
```

Comment:
I don't think this is necessary given that identity must have a type with value "SystemAssigned" ?

Reply:
Yes. But it is hard to track this case.